test/unit/SpatialReferenceTest.cpp

```diff
@@ -316,6 +316,59 @@ TEST(SpatialReferenceTest, test_vertical_and_horizontal)
 
 }
 
+TEST(SpatialReferenceTest, readerOptions)
+{
+    const std::string utm16(SpatialReference("EPSG:26916").getWKT());
+    std::string native;
+
+    {
+        Options o;
+        o.add("filename", Support::datapath("las/test_utm17.las"));
+        LasReader r;
+        r.setOptions(o);
+
+        PointTable t;
+        r.prepare(t);
+        native = r.getSpatialReference().getWKT();
```

Comment:
This depends to some extent on the code that you've added working as you expect.  It seems worthwhile to check this against the known SRS of the file.

Reply:
Updated this to use the QuickInfo output.